src/transcript-plugin.tsx

```diff
@@ -78,27 +81,39 @@ export class TranscriptPlugin extends KalturaPlayer.core.BasePlugin {
   }
 
   private _initListeners(): void {
-    this.eventManager.listen(this.player, this.player.Event.FIRST_PLAYING, () => {
-      if ((this.player.getTracks(this.player.Track.TEXT) || []).length) {
-        // check if captions already added in TextTrack
-        if (!this._captionMap.size) {
-          // turn on loading animation till captions added to TextTrack
-          this._isLoading = true;
-        }
+    this.eventManager.listenOnce(this.player, this.player.Event.TRACKS_CHANGED, () => {
+      if (this._getTextTracks().length) {
+        this.eventManager.listen(this.player, this.player.Event.TIMED_METADATA_CHANGE, this._onTimedMetadataChange);
+        this.eventManager.listen(this.player, this.player.Event.TIMED_METADATA_ADDED, this._onTimedMetadataAdded);
+        this.eventManager.listen(this.player, this.player.Event.TEXT_TRACK_CHANGED, this._handleLanguageChange);
         this._addDownloadIcon();
         this._addPrintIcon();
         this._addTranscriptItem();
+        this._initLoading();
       }
     });
-    this.eventManager.listen(this.player, this.player.Event.TIMED_METADATA_CHANGE, this._onTimedMetadataChange);
-    this.eventManager.listen(this.player, this.player.Event.TIMED_METADATA_ADDED, this._onTimedMetadataAdded);
-    this.eventManager.listen(this.player, this.player.Event.TEXT_TRACK_CHANGED, this._handleLanguageChange);
   }
 
+  private _initLoading = () => {
+    clearTimeout(this._loadingTimeoutId);
```

Comment:
is it possible for `_loadingTimeoutId` to be null or undefined at this point? if so, how would it affect clearTimeout?

Reply:
check in console
`clearTimeout(null);
clearTimeout(undefined);
clearTimeout(() => {});
clearTimeout();`
you'll see that param can have any type and not required.

Reply:
cool, thanks